src/components/Admin/Collection/Collection.js

```diff
@@ -0,0 +1,101 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+import DataController from '../utils/DataController';
+
+import WebcamCapture from '../../WebcamCapture/WebcamCapture';
+import ItemSelector from '../ItemSelector';
+
+class Collection extends Component {
+  state = {
+    item: 'unknown',
+    burstCount: '10',
+    status: 'Loading...',
+    busy: true
+  };
+
+  items = [];
+  webcamCapture = React.createRef();
+
+  burstShot = () => {
+    let counter = this.state.burstCount;
+    const ticker = setInterval(async () => {
+      this.setState({
+        busy: true,
+        status:
+          `Capturing image ${this.state.burstCount - counter}` +
+          `/${this.state.burstCount}`
+      });
+
+      const img = await this.webcamCapture.current.requestScreenshot();
+      this.dataController.addImage(img.src, this.state.item);
+
+      counter--;
+      if (counter <= 0) {
+        clearInterval(ticker);
+        this.setState({busy: false, status: 'Done'});
+      }
+    }, 1000);
+  };
+
+  componentDidMount() {
+    this.dataController = new DataController();
+    this.dataController.getStoreList().then(items => {
+      this.items = Object.values(items);
+      this.setState({busy: false, status: 'Ready'});
+    });
+  }
+
+  back = () => {
+    this.props.history.replace('/admin');
+  };
+
+  render() {
+    return (
+      <div className="page">
+        <WebcamCapture
+          ref={this.webcamCapture}
+          imgSize={224}
+          onFail={() => {}}
```

Comment:
do nothing onFail? this is kindof pointless

Reply:
It is but the prop is required and with it being an admin page I wasn't bothered about gracefully handling the issue, it just won't render a webcam.

Reply:
why is it required - its obviously not needed in this case?